core/unit_test/TestGraph.hpp

```diff
@@ -403,6 +405,107 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), empty_graph) {
 }
 
+template <typename ViewType>
+struct ForceGlobalLaunchFunctor {
+ public:
+  static constexpr size_t count =
+#if defined(KOKKOS_ENABLE_CUDA)
+      Kokkos::Impl::CudaTraits::ConstantMemoryUsage +
+#elif defined(KOKKOS_ENABLE_HIP)
+      Kokkos::Impl::HIPTraits::ConstantMemoryUsage +
+#endif
+      1;
+
+  ViewType data;
+
+  ForceGlobalLaunchFunctor(ViewType data_) : data(std::move(data_)) {}
+
+  template <typename T>
+  KOKKOS_FUNCTION void operator()(const T) const {
+    ++data();
+  }
+
+ private:
+  std::byte unused[count] = {};
+};
+
+// Ensure that that "global memory launch" path works.
+TEST_F(TEST_CATEGORY_FIXTURE(graph), force_global_launch) {
+#if defined(KOKKOS_ENABLE_CUDA)
+  if constexpr (!std::is_same_v<TEST_EXECSPACE, Kokkos::Cuda>) {
+#elif defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_IMPL_HIP_NATIVE_GRAPH)
+  if constexpr (!std::is_same_v<TEST_EXECSPACE, Kokkos::HIP>) {
+#endif
+    GTEST_SKIP() << "This execution space does not support global launch.";
+
+#if defined(KOKKOS_ENABLE_CUDA) || \
+    (defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_IMPL_HIP_NATIVE_GRAPH))
+  }
+  using value_t   = int;
+  using view_t    = Kokkos::View<value_t, TEST_EXECSPACE,
+                              Kokkos::MemoryTraits<Kokkos::Atomic>>;
+  using functor_t = ForceGlobalLaunchFunctor<view_t>;
+
+  const std::string kernel_name = "Let's make it a huge kernel";
+  const std::string alloc_label =
+      kernel_name + " - GraphNodeKernel global memory functor storage";
+
+  view_t data(Kokkos::view_alloc("witness", ex));
+
+  using namespace Kokkos::Test::Tools;
+  listen_tool_events(Config::DisableAll(), Config::EnableAllocs());
+
+  std::optional<Kokkos::Experimental::Graph<TEST_EXECSPACE>> graph;
+
+  const void* ptr   = nullptr;
+  uint64_t ptr_size = 0;
+
+  ASSERT_TRUE(validate_existence(
+      [&]() {
+        graph = Kokkos::Experimental::create_graph(ex, [&](const auto& root) {
+          auto node = root.then_parallel_for(
+              kernel_name,
+              Kokkos::Experimental::require(
+                  Kokkos::RangePolicy<TEST_EXECSPACE>(0, functor_t::count),
+                  Kokkos::Experimental::WorkItemProperty::HintHeavyWeight),
+              functor_t(data));
+        });
+      },
+      [&](AllocateDataEvent alloc) {
+        if (alloc.name != alloc_label)
+          return MatchDiagnostic{
+              false, {"Allocation name mismatch (got " + alloc.name + ')'}};
+        if (alloc.size < functor_t::count)
+          return MatchDiagnostic{
+              false,
+              {"Allocation size mismatch (expected at least " +
+               std::to_string(functor_t::count) + " but got " +
+               std::to_string(alloc.size) + ')'}};
+        ptr      = alloc.ptr;
+        ptr_size = alloc.size;
+        return MatchDiagnostic{true};
+      }));
+
+  graph->instantiate();
+  TEST_EXECSPACE{}.fence(
+      "Ensure that kernel dispatch to global memory is finished "
+      "before submission.");
```

Comment:
Please confirm you meant to fence the default instance and not `ex`

Reply:
Yes, I confirm.

The reason is that the policy is created with:
```c++
Kokkos::RangePolicy<TEST_EXECSPACE>(0, functor_t::count)
```
so it should be using the default execution space instance for the kernel dispatching to global memory, right ?

Reply:
I think this is worth a comment.

Reply:
Done :heavy_check_mark: